daemon/image_delete.go

```diff
@@ -88,7 +88,7 @@ func (daemon *Daemon) DeleteImage(eng *engine.Engine, name string, imgs *engine.
 		return nil
 	}
 
-	if len(repos) <= 1 {
+	if len(repos) <= len(tags) {
```

Comment:
Looking for some input on this. Ultimately, the images get deleted if, *after* untagging everything in the `tags` array - there are no tags left associated with the image. So, the check for whether or not the image can be deleted should be made if the end state will be such that there are no remaining tags left on the image (or, ultimately, the same variable that drives this `if` statement should also drive the `if` around the actual image deletion.